packages/cli/src/start/index.js

```diff
@@ -36,14 +61,48 @@ async function start({flags}) {
   }
 }
 
-function selectPort(flags) {
-  if (!isNaN(Number(flags.port))) {
-    return Number(flags.port);
+function schemaFromOptions(ops) {
+  if (ops.privateKey && ops.certChain) {
+    return 'https';
+  }
+  return 'http'
+}
+
+function selectNumberFromKeys(keys, def) {
+  const number = keys.map(i => Number(i)).find(i => !isNaN(i));
+  if (number > 0) {
+    return number;
   }
+  return def;
+}
 
-  if (!isNaN(Number(process.env.PORT))) {
-    return (process.env.PORT);
+function selectStringFromKeys(keys, def) {
+  const str = keys.find(i => typeof(i) == "string");
+  if (str) {
+    return str;
   }
+  return def;
+}
+
 
-  return 1337;
+function readFileFromKeys(keys, spinner, def) {
+  return new Promise((resolve, _) => {
+    const fname = selectStringFromKeys(keys, def);
+    if (fname) {
```

Comment:
Split `readFileFromKeys` into two distinct functions: `selectPathFromKeys`, then read from the selected path. Throw if the path is missing or the provided file can not be read. The current implementation drops to `http` if a cert/key is not found.